fire/api/firedb.py

```diff
@@ -89,6 +89,19 @@ def hent_punkt(self, ident: str) -> Punkt:
 
         return self._cache["punkt"][ident]
 
+    def hent_punkt_liste(self, identer: List[str]) -> List[Punkt]:
+        """
+        Returnerer en liste af punkter der matcher identerne i listen `identer`.
+        """
+        punkter = []
+        for ident in identer:
+            try:
+                punkter.append(self.hent_punkt(ident))
+            except NoResultFound:
+                raise ValueError(f"Ident {ident} ikke fundet i databasen")
```

Comment:
1. Ville det ikke være mere i tråd med resten af systemet at re-raise en `NoResultFound` med samme beskrivende tekst som du her putter ind i en anden slags undtagelse? Så `NoResultFound` er det eneste man behøver checke for?

2. Det ville være nyttigt med en valgfri parameter `ignorer_ukendte=False`, så man kan vælge bare at få de elementer fra inputlisten, der kan findes: Det ville for MTLs vedkommende (muligvis) gøre at man kan kaste `alle_punkter` efter `hent_punkt_liste` uden at spekulere over at der indgår nyetablerede, endnu ikke registrerede, punkter

Reply:
Jeg var lidt i tvivl om den bedste fremgangsmåde her. Jeg gættede på at det nok var bedst at få at vide hvis man leder efter et punkt der ikke findes, men det kan være det ikke er nødvendigt. Det jeg tænkte var, at man nok alligevel så bagefter ville sammenligne sin ident-liste med de punkter der kommer ud af funktionen, og ja, så kan funktionen jo lige så godt gøre arbejdet up-front. 

`hent_observationer()` returnerer bare de observationer der kan findes fra input-listen uden at brokke sig. Det har jeg studset over tidligere og spekuleret på om var smart eller ej. Jeg har ikke fundet det rigtige svar endnu, men det kunne nok være smart om de to funktioner opfører sig nogenlunde ens.

1. Jo, det kunne det også godt. Men da problemet jo i virkeligheden stammer fra dårligt input valgte jeg `ValueError` som jo betyder at du har et problem i dit input. Og så kan man jo sige at det ikke nødvendigvis er "No result found", da det sikkert kun er et enkelt ident ud af mange der ikke kan findes. Måske mere `SomeResultsNotFound` :-)

2. Måske skal det nærmere være 

```
def hent_punkt_liste(self, identer: List[str], ignore_ukendte=True) -> List[Punkt]:
```

da den så som udgangspunkt opfører sig ligesom `hent_observationer()`. Så kan man altid slå det fra hvis man har brug for mere kontrol.

Reply:
> Men da problemet jo i virkeligheden stammer fra dårligt input valgte jeg ValueError som jo betyder at du har et problem i dit input

Uafviseligt argument :-)

> da den så som udgangspunkt opfører sig ligesom hent_observationer(). Så kan man altid slå det fra hvis man har brug for mere kontrol.

God pointe - jeg har ikke en personlig preference for den ene eller den anden vej, men jeg går stærkt ind for at den er symmetrisk med `hent_observationer()`.

Så det må være
```python
def hent_punkt_liste(self, identer: List[str], ignorer_ukendte: bool = True) -> List[Punkt]:
```
(med typeannotering og r i ignore**r** :-)

Reply:
(og så glemte jeg den futuristiske typeannotering på `self` )

Reply:
> Så det må være

Så gør vi sådan!

> (og så glemte jeg den futuristiske typeannotering på self )

Er det ikke bare støj? Der kan jo ingen tvivl være om hvilken type `self` er. Det ser heller ikke ud til at være en praksis [PEP 563](https://www.python.org/dev/peps/pep-0563/) ligger op til skal udføres.

Reply:
Så skulle den være på plads.

Reply:
> Er det ikke bare støj? Der kan jo ingen tvivl være om hvilken type self er.

Det mener jeg også